src/lib/fields/VocabularyField.js

```diff
@@ -87,3 +87,106 @@ export class AllowAdditionsVocabularyField extends VocabularyField {
     );
   }
 }
+
+/**
+ * Serialize and deserialize rights field that can contain vocabulary values
+ * and free text but sharing structure with the vocabulary values
+ */
+export class RightsVocabularyField extends VocabularyField {
+  constructor({
+    fieldpath,
+    deserializedDefault = null,
+    serializedDefault = null,
+    localeFields = [],
+  }) {
+    super({ fieldpath, deserializedDefault, serializedDefault });
+    this.localeFields = localeFields;
+  }
+
+  /**
+   * Deserializes the values in the format:
+   * {id: 'vocab_id'} for controlled vocabs and
+   * {<field_name>: 'field_name', <field_descripton>: 'field_descripton', ...}
+   * for user added entries
+   *
+   * @param {Object} record - Record to deserialize
+   * @param {String} defaultLocale - The default locale
+   * @returns
+   */
+  deserialize(record, defaultLocale) {
+
+    const fieldValue = _get(record, this.fieldpath, this.deserializedDefault);
+    const _deserialize = (value) => {
+      if ('id' in value) {
+        if (typeof value.title === 'string') {
+          // Needed in case we pass a default value
```

Comment:
Needed for [this](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/config.py#L495-L509). Ideally we would just add the id as default and retrieve the value in the UIObject but it's not really working because we don't have a record created and we only serialize it when we edit (check [here](https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/views/deposits.py#L224-L225)).